modules/menu/menu.c

```diff
@@ -544,6 +545,40 @@ static int hold_prev_call(struct re_printf *pf, void *unused)
 	return call_hold(ua_prev_call(uag_cur()), true);
 }
 
+static int switch_audio_dev(struct re_printf *pf, void *arg)
+{
+	const struct cmd_arg *carg = arg;
+	static bool switch_aud_inprogress;
+	char *mod;
+	char *device;
+	int err = 0;
+
+	if (!switch_aud_inprogress && !carg->complete) {
+		statmode = STATMODE_OFF;
+		re_hprintf(pf, "\rPlease enter transfer target SIP uri:\n");
+	}
+
+	switch_aud_inprogress = true;
+
+	if (carg->complete) {
+
+		mod = strtok( carg->prm, ",");
+		device = strtok( NULL, ",");
+		if( mod == NULL || device == NULL){
+			re_hprintf(pf, "\rFormat should be: alsa,default\n");
+			return 0;
+		}
+
+		err = audio_set_player(call_audio(ua_call(uag_cur())), mod, device);
+		if( err)
+			return err;
+
+		err = audio_set_source(call_audio(ua_call(uag_cur())), mod, device);
```

Comment:
instead of fetching the struct audio from uag_cur twice, you can move that
operation to the top so that it is only done once (minor detail)